torch/jit/frontend.py

```diff
@@ -286,14 +287,31 @@ def build_class_def(ctx, py_def, methods, properties, self_name):
 
 
 def build_def(ctx, py_def, type_line, def_name, self_name=None):
+    global _args_and_types  # type: ignore[name-defined]
```

Comment:
Why does it need to be `global`? it seems we are assigning a value to it on line 303

Reply:
Somehow without making `_args_and_types` as `global` it doesn't get recognized.

Reply:
Here is the error,
```
NameError: name '_args_and_types' is not defined
```